limiter.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"sort"
+	"strconv"
+	"time"
+)
+
+type (
+	// Config provides configuration values for creating a new rate-limiter.
+	Config struct {
```

Comment:
newer golang code typically eschews this config struct pattern in favor of making required information positional arguments and specifying optional information with functions. more info on this pattern: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

in this case it seems like `Redis` and `Buckets` would be positional and the others would be options